internal-log-fetcher/src/discovery.rs

```diff
@@ -115,10 +144,7 @@ async fn discover_aws(aws: &AwsConfig) -> Result<HashSet<NodeIdentity>> {
     });
 
     for (_, meta) in aws_results {
-        let colon_ix = meta
-            .remote_addr
-            .find(':')
-            .unwrap_or_else(|| meta.remote_addr.len());
+        let colon_ix = meta.remote_addr.find(':').unwrap_or(meta.remote_addr.len());
```

Comment:
Perhaps it's worth using the same `node_identity` helper function for uniformity?